service/health/types.go

```diff
@@ -98,3 +130,25 @@ func nodeMemoryToInt(nodeMemory *resource.Quantity) int64 {
 	}
 	return memBytes
 }
+
+func nodeHasCondition(conditions []v1.NodeCondition, s v1.ConditionStatus, t v1.NodeConditionType) bool {
+	for _, c := range conditions {
+		if c.Status == s && c.Type == t {
+			return true
+		}
+	}
+
+	return false
+}
+
+func calculateNodeHealth(node v1.Node) key.Health {
+	ready := nodeHasCondition(node.Status.Conditions, v1.ConditionTrue, v1.NodeReady)
+	if ready {
+		return key.Green
+	}
+	role := nodeRole(node.ObjectMeta.Labels)
+	if role == "master" {
+		return key.Red
+	}
+	return key.Yellow
```

Comment:
Why is it yellow when it isn't master? Does it mean worker can never be red?

Reply:
That's how we originally speced it out when we were in Rome. It felt a little strange to me when I was implementing it. Maybe we should revisit these decisions soon.